### DISCUSSION TITLE
How to solve coc-clangd error “clang pp_file_not_found”

### DISCUSSION THREAD

**Discussion by Sipym** [neoclide/coc.nvim#4286] [Q&A] [answered]
When I was writing a project about C language, I found that coc-clangd always reported an error "",and the project directory structure is as follows,
![image](https://user-images.githubusercontent.com/94687431/195621080-1f2fe1a7-c489-4d78-9460-146d9f4f9c7d.png)
This is the error message:
![image](https://user-images.githubusercontent.com/94687431/195621226-733757a0-57b6-4c70-9dd7-3b1fb84f60cc.png)

I don't know how to get clangd to get the header file information of my project. Is there any solution?
Thank you for your reply.

**Comment by fannheyward** [comment, marked as answer]
You need to setup `compile_commands.json` for clangd.

**Reply by Sipym** [reply]
Thank you,i have solved this problem.
